### PR TITLE
package/QGIS: fix cmake/FindPyqt.py

### PR DESCRIPTION
**Pull request by Sinan81**
QGIS cmake file needs to be fixed due to new sip set up shown in PR https://github.com/spack/spack/pull/15297

@adamjstewart

### CODE REVIEW

**Comment by Sinan81** [comment]
Before:
```
$ python ./cmake/FindPyQt5.py 
pyqt_version:050d00
pyqt_version_num:331008
pyqt_version_str:5.13.0
pyqt_version_tag:
pyqt_mod_dir:/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-7.4.0/python-3.7.6-iegbz4qm54txwdabczzic5pds5ualjwc/lib/python3.7/site-packages/PyQt5
pyqt_sip_dir:/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-7.4.0/py-sip-4.19.18-shc42xd2jp3xnw6wanywykmgdwwsqtu7/share/sip
pyqt_sip_flags:-n PyQt5.sip -t WS_X11 -t Qt_5_14_0
pyqt_bin_dir:/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-7.4.0/python-3.7.6-iegbz4qm54txwdabczzic5pds5ualjwc/bin
pyqt_sip_module:PyQt5.sip
```
After
```
$ python ./FindPyQt5.py 
pyqt_version:050d01
pyqt_version_num:331009
pyqt_version_str:5.13.1
pyqt_version_tag:
pyqt_mod_dir:/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-7.4.0/py-pyqt5-5.13.1-x3nq42q46thpuurle5iybc46rhwt24yv/lib/python3.7/site-packages/PyQt5
pyqt_sip_dir:/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-7.4.0/py-pyqt5-5.13.1-x3nq42q46thpuurle5iybc46rhwt24yv/share/sip/PyQt5
pyqt_sip_flags:-n PyQt5.sip -t WS_X11 -t Qt_5_12_4
pyqt_bin_dir:/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-7.4.0/py-pyqt5-5.13.1-x3nq42q46thpuurle5iybc46rhwt24yv/bin
pyqt_sip_module:PyQt5.sip
```

**Comment by adamjstewart** [comment]
You need to update the `package.py` to call the patch

**Comment by Sinan81** [comment]
Ooops

File: var/spack/repos/builtin/packages/qgis/pyqt.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
+diff --git a/cmake/FindPyQt5.py b/cmake/FindPyQt5.py
+index a42cc01..8d71e9d 100644
+--- a/cmake/FindPyQt5.py
++++ b/cmake/FindPyQt5.py
+@@ -40,6 +40,7 @@ except ImportError:
+     import sys
+     cfg = sipconfig.Configuration()
+     sip_dir = cfg.default_sip_dir
++    pyqt_prefix = '/'.join(PyQt5.QtCore.__file__.split('/')[0:-5])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qgis/pyqt.patch]: Use `os.path.join` and `os.sep`, not all OSes use `/`

File: var/spack/repos/builtin/packages/qgis/pyqt.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+@@ -54,9 +55,9 @@ except ImportError:
+         'pyqt_version': PyQt5.QtCore.PYQT_VERSION,
+         'pyqt_version_str': PyQt5.QtCore.PYQT_VERSION_STR,
+         'pyqt_sip_flags': PyQt5.QtCore.PYQT_CONFIGURATION['sip_flags'],
+-        'pyqt_mod_dir': os.path.join(cfg.default_mod_dir, "PyQt5"),
+-        'pyqt_sip_dir': sip_dir,
+-        'pyqt_bin_dir': cfg.default_bin_dir,
++        'pyqt_mod_dir': os.path.dirname(PyQt5.QtCore.__file__),
++        'pyqt_sip_dir': pyqt_prefix+'/share/sip/PyQt5',
++        'pyqt_bin_dir': pyqt_prefix+'/bin',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qgis/pyqt.patch]: Same here

File: var/spack/repos/builtin/packages/qgis/pyqt5.patch [hunk 3]
```diff
@@ -0,0 +1,25 @@
+diff --git a/cmake/FindPyQt5.py b/cmake/FindPyQt5.py
+index 6a55a0f801..d6eda1fa1e 100644
+--- a/cmake/FindPyQt5.py
++++ b/cmake/FindPyQt5.py
+@@ -40,6 +40,7 @@ except ImportError:
+     import sys
+     cfg = sipconfig.Configuration()
+     sip_dir = cfg.default_sip_dir
++    pyqt_prefix = os.sep.join(PyQt5.QtCore.__file__.split(os.sep)[0:-5])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qgis/pyqt5.patch]: os.path.join

Comment (Sinan81) [reply var/spack/repos/builtin/packages/qgis/pyqt5.patch]:
```
In [1]: import PyQt5.QtCore

In [7]: PyQt5.QtCore.__file__.split(os.sep)[0:-2]
Out[7]: ['', 'usr', 'lib', 'python2.7', 'dist-packages']

In [9]: os.path.join(PyQt5.QtCore.__file__.split(os.sep)[0:-2])
Out[9]: ['', 'usr', 'lib', 'python2.7', 'dist-packages']

In [10]: os.path.join(*PyQt5.QtCore.__file__.split(os.sep)[0:-2])
Out[10]: 'usr/lib/python2.7/dist-packages'

In [12]: os.sep.join(PyQt5.QtCore.__file__.split(os.sep)[0:-2])
Out[12]: '/usr/lib/python2.7/dist-packages'
```
Above note the missing leading `/` when we do `os.path.join`. So `os.sep.join` is the right choice it seems.

Comment (Sinan81) [reply var/spack/repos/builtin/packages/qgis/pyqt5.patch]: or the other option is:
```
In [6]: os.path.join(os.sep,*PyQt5.QtCore.__file__.split(os.sep)[0:-2])
Out[6]: '/usr/lib/python2.7/dist-packages'
```

**Pull request merged**